### PR TITLE
Improve docs.rs builds

### PR DESCRIPTION
**Pull request by novacrazy**
While working on [`generic-array 1.0`](https://docs.rs/generic-array/1.0.0-alpha.1/), I noticed docs.rs does not contain the `const-generics` portions of `typenum`, so the links were broken. Enabling support for that with automatic "Available on crate feature `const-generics` only." text would have been easy, except for the `force_unix_path_separator` feature interfering with it. 

After looking through the commit history, your original commit and message 694130d80cee6316879b1eb4889a7db1de3755b3 implied _enabling_ the feature made it more compatible. If I'm wrong, please correct me, but I don't see any reason why this should not be the default behavior. Making `force_unix_path_separator` the default behavior has no visible downsides and allows `doc_auto_cfg` to function correctly, more or less. 

I added a couple extra hints to rustdoc for i128 implementations as well. Unfortunately, these rustdoc hints don't work for the `scale-info` derives, so they are still excluded from docs.rs.

As a bonus, I added a Rust Playground metadata section to enable i128 and const-generics support there.

### CODE REVIEW

**Comment by novacrazy** [comment]
@paholg I'm looking to get `generic-array 1.0` published as soon as possible, but with typenum not documenting the `const-generics` feature, docs.rs links will be broken. Could this PR be looked at? Or delegate it to another maintainer.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ The MSRV (Minimum Supported Rust Version) is 1.37.0, and typenum is tested
 against this Rust version.
 
 ### Unreleased
+- [removed] Remove `force_unix_path_separator` feature, make it the default
```
Comment (paholg) [inline CHANGELOG.md:9]: Why is this removed?

IIRC, there was a reason for having it as optional. I think folks had issues both with it and without it, in different environments. I think it was related to cross-compilation involving Windows and/or Bazel?

**Review by paholg — COMMENTED** [review]
Sorry for the delay. Just want to make sure we're not breaking anything for anyone with removal of `force_unix_path_separator` and I'll be happy to merge/publish.

File: CHANGELOG.md (hunk 1 above)
Comment (novacrazy) [reply CHANGELOG.md:9]: The original commit (694130d80cee6316879b1eb4889a7db1de3755b3) suggested that _enabling_ `force_unix_path_separator` improved the compatibility with non-Cargo build tools. It looks like the previous version only used the environment variables emitted by cargo to do the including. Effectively, `force_unix_path_separator` is now the default, with no environment variables, which is how every crate I've ever seen does things.

Furthermore, it interfered with the `doc_auto_cfg` feature. Instead of picking up the `const-generics` crate feature, it would use `force_unix_path_separator`, which was misleading and ugly.

Comment (paholg) [reply CHANGELOG.md:9]: Did some digging.

Looks like Bazel can't process the custom environment variables: https://github.com/paholg/typenum/issues/127, so this improves the situation there.

However, it seems the solution without env variables breaks wasm-pack on windows: https://github.com/paholg/typenum/issues/124

Maybe wasm-pack works better now? I don't have a Windows dev machine, but I'll do some more digging and try to figure out if this is safe to remove.

Comment (novacrazy) [reply CHANGELOG.md:9]: I've most certainly used wasm-pack in conjunction with this form of including in other crates. Will try to reproduce with typenum today. Includes are purely a rustc thing, too, not sure how it would affect wasm-pack.

However, don't let this hold up publishing a new version with the 32-bit build fixes. I've already published `generic-array` 1.0, with links in the readme/crate that link back to our Github Pages docs that includes typenum with `const-generics`, since the docs.rs build was broken.

Comment (novacrazy) [reply CHANGELOG.md:9]: I'm on a Windows host, using `wasm-pack new test-project`

Adding to `Cargo.toml`:
```toml
[patch.crates-io]
typenum = { git = "https://github.com/novacrazy/typenum" }
```

then adding some code that uses `generic-array 0.14`, running `wasm-pack build` builds/runs totally fine. If I open up the same project on a Linux VM, it also builds totally fine. 

I'm unsure if there ever was an actual issue?

Comment (paholg) [reply CHANGELOG.md:9]: Hm, okay. Let's ship it!

Comment (novacrazy) [reply CHANGELOG.md:9]: It's possible Rust itself updated at some point to fix paths automatically since 2018, perhaps https://github.com/rust-lang/rust/pull/89174/ in 1.58.0 in 2022, or some other minor fix somewhere.

**Comment by MingweiSamuel** [comment]
This just breaks the build on windows?

#107
#108
#125

**Comment by paholg** [comment]
> This just breaks the build on windows?

Can you be more specific? Typenum's CI runs builds on Windows, so there are at least some cases where it doesn't.

**Comment by MingweiSamuel** [comment]
Yes, seems it only occurs in a situation where `"\\?\"` gets prepended to `OUT_DIR` env var. Not sure why that is getting appended, but testing shows that it is causing the error:

```rust
mod gen1 {
    include!(r"D:\Projects\typenum_test/include.rs");
}

mod gen2 {
    include!(r"\\?\D:\Projects\typenum_test\include.rs");
}

mod gen3 {
    include!(r"\\?\D:\Projects\typenum_test/include.rs");
}
```

first two work, but the last causes an error

**Comment by MingweiSamuel** [comment]
It's this specific issue: https://github.com/rust-lang/rust/issues/75075#issuecomment-667733716

Error happens if `OUT_DIR` path is prefixed with `\\?\` in which case `/` no longer works as a path separator on windows

**Comment by novacrazy** [comment]
I do actually kind of like the way that hydroflow PR handled the dynamic include path. That likely wouldn’t interfere with the docs.rs stuff like the previous system, which was the point of this PR. 

Still, the issue as you described this is a pervasive issue with Windows/Rust in general and to patch it in a few specific crates is a bandaid at best.

**Comment by MingweiSamuel** [comment]
I will merge a mitigation fix in hydroflow to remove the `\\?\` prefix which will allow it to work with `/` paths (when possible). I can also make a PR for this crate to re-introduce the per-platform path separator logic a la the hydroflow PR if you want

**Comment by paholg** [comment]
@MingweiSamuel I would welcome a PR.

My concern is if it falls to the same issues that my prior attempts had. Ideally, we would reproduce the various problematic setups in CI to ensure things work for everyone, but I'm not sure how realistic that is.
